Documentation/Debugging.md

```diff
@@ -0,0 +1,46 @@
+# Implementation Validation with Assurance
+
+The AEP SDK offers an extension to inspect quickly, validate, debug data collection, and experiences for any mobile app using the AEP SDK. We built Assurance to do the heavy lifting of getting an SDK implementation right, so app developers can focus on creating engaging experiences.
+
+## Quick Setup
+
+#### Install Compatibility Layer
+
+The Assurance extension requires the backward-compatible layer. Instructions on installing the backward compatible layer can be found [here](./Migration/ACP-Migration.md).
+
+#### Add Assurance to your app
+
+Include the Assurance extension via Cocoapods.
+
+```ruby
+pod 'AEPAssurance'
+```
+
+#### Update Extension Registration
+
+Now that the backward compatible layer is installed, `ACPCore` can now be imported into the `AppDelegate` file allowing Assurance to be registered. Replace usage of `MobileCore.registerExtensions` with `ACPCore.registerExtensions` while including the Assurance extension.
+
+```diff
++ import ACPCore
++ import AEPAssurance
+
+func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
+
+-  MobileCore.registerExtensions([Lifecycle.self, Identity.self, Signal.self, ...], {
+-   //...
+-  })  
+
++  ACPCore.registerExtensions([Lifecycle.self, Identity.self, Signal.self, AEPAssurance.self, ...], {
```

Comment:
does this really work? I don't think `AEPAssurance` is an extension class

Reply:
mmmm. good catch, yeah I forgot it doesn't directly inherit from `ACPExtension`. let me fix this.

Reply:
Will also fix in my 3rd party PR..

Reply:
https://github.com/adobe/aepsdk-sample-app-ios/blob/main/Swift/AEPSampleApp/AppDelegate.swift#L54-L57
This is the code we have in the sample app

Reply:
looks like we need to update the sample app.

Reply:
that was the only way I can get Assurance working. `start` method has to be called.

Reply:
I think with this change: https://github.com/adobe/aepsdk-compatibility-ios/pull/5 it allows a more standard approach now.

Reply:
I'll give this a test and update the sample app if needed.

Reply:
I see, it does work. Didn't notice this change.
But still, it is something quite confusing.

Reply:
I do agree it can be confusing, however, it's quite a big improvement from having to register the extension in the callback then start the SDK again.

Given the only change on this workflow is using ACPCore instead of MobileCore on registerExtensions, it would be optimal if we could somehow provide the backward compatible API under MobileCore. I attempted to create an Obj-C category on MobileCore in the compatibility layer, but this causes a circular dependency which caused compiling to fail when included from Cocoapods. Maybe we can take another swing at this.